packages/react-native-editor/src/index.js

```diff
@@ -1,161 +1,83 @@
- * WordPress dependencies
+ *	Register Gutenberg editor to React Native App registry.
+ *
+ * @typedef {Object} PluginTranslation
+ * @property {string}              domain                       Domain of the plugin.
+ * @property {Function}            getTranslation               Function for retrieving translations for a locale.
+ *
+ * @param    {Object}              arguments
+ * @param    {Function}            arguments.beforeInitCallback Callback executed before the editor initialization.
+ * @param    {PluginTranslation[]} arguments.pluginTranslations Array with plugin translations.
  */
-import {
-	validateThemeColors,
-	validateThemeGradients,
-} from '@wordpress/block-editor';
-import { unregisterBlockType, getBlockType } from '@wordpress/blocks';
-
-const reactNativeSetup = () => {
-	// Disable warnings as they disrupt the user experience in dev mode
-	// eslint-disable-next-line no-console
-	console.disableYellowBox = true;
-
-	I18nManager.forceRTL( false ); // Change to `true` to debug RTL layout easily.
-};
-
-const gutenbergSetup = () => {
-	const wpData = require( '@wordpress/data' );
-
-	// wp-data
-	const userId = 1;
-	const storageKey = 'WP_DATA_USER_' + userId;
-	wpData.use( wpData.plugins.persistence, { storageKey } );
-
-	setupApiFetch();
-
-	const isHermes = () => global.HermesInternal !== null;
-	// eslint-disable-next-line no-console
-	console.log( 'Hermes is: ' + isHermes() );
-
-	setupInitHooks();
-
-	const initializeEditor = require( '@wordpress/edit-post' ).initializeEditor;
-	initializeEditor( 'gutenberg', 'post', 1 );
-};
-
-const setupInitHooks = () => {
-	const wpHooks = require( '@wordpress/hooks' );
-
-	wpHooks.addAction(
-		'native.pre-render',
-		'core/react-native-editor',
-		( props ) => {
-			setupLocale( props.locale, props.translations );
-
-			const capabilities = props.capabilities ?? {};
-			if (
-				getBlockType( 'core/block' ) !== undefined &&
-				capabilities.reusableBlock !== true
-			) {
-				unregisterBlockType( 'core/block' );
-			}
-		}
-	);
-
-	// Map native props to Editor props
-	// TODO: normalize props in the bridge (So we don't have to map initialData to initialHtml)
-	wpHooks.addFilter(
-		'native.block_editor_props',
-		'core/react-native-editor',
-		( props ) => {
-			const { capabilities = {} } = props;
-			let {
-				initialData,
-				initialTitle,
-				postType,
-				featuredImageId,
-				colors,
-				gradients,
-				rawStyles,
-				rawFeatures,
-				galleryWithImageBlocks,
-				locale,
-			} = props;
-
-			if ( initialData === undefined && __DEV__ ) {
-				initialData = initialHtml;
-			}
-			if ( initialTitle === undefined ) {
-				initialTitle = 'Welcome to Gutenberg!';
-			}
-			if ( postType === undefined ) {
-				postType = 'post';
+const registerGutenberg = ( {
+	beforeInitCallback,
+	pluginTranslations = [],
+} = {} ) => {
+	class Gutenberg extends Component {
+		constructor( props ) {
+			super( props );
+
+			const parentProps = omit( this.props || {}, [ 'rootTag' ] );
```

Comment:
```suggestion
			const { rootTag, ...parentProps } = this.props;
```
While I know `lodash` is ubiquitous, I wonder if we should attempt to leverage language tooling rather than a third-party library. WDYT? 

Additionally, I do not believe the fallback `{}` is required, as `this.props` will default to an empty object.

Reply:
> While I know lodash is ubiquitous, I wonder if we should attempt to leverage language tooling rather than a third-party library. WDYT?

Yep I agree, we could leverage the same functionality with your suggestion. Btw, I found [this useful repo](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_omit) with alternatives to Lodash functions.

> Additionally, I do not believe the fallback {} is required, as this.props will default to an empty object.

Good point, I also do think it's not required as, AFAIK, React guarantees that the props object is always defined ([reference](https://reactjs.org/docs/react-component.html#constructor)).

Reply:
I applied the suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/37073/commits/1f7cd1eac817c078f22c0d8c84b0a1b33716be50).